WWDCFamily/Source/Auth/Controllers/AuthViewController.swift

```diff
@@ -25,10 +32,36 @@ class AuthViewController: UIViewController {
     override var preferredStatusBarStyle: UIStatusBarStyle {
         return .lightContent
     }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        self.authStateDidChangeHandle = self.auth?.addStateDidChangeListener(self.updateUI(auth:user:))
```

Comment:
We probably don't want this logic in viewWillAppear and viewWillDissapear .

We should also have an abstraction around user login state changing rather then in-line in a VC